requests/cookies.py

```diff
@@ -144,11 +144,10 @@ def remove_cookie_by_name(cookiejar, name, domain=None, path=None):
     """
     clearables = []
     for cookie in cookiejar:
-        if cookie.name != name:
-            continue
-        if domain is not None and domain != cookie.domain:
-            continue
-        if path is not None and path != cookie.path:
+        bool_cookie = (cookie.name != name)
```

Comment:
@kennethreitz : I read about this anti-pattern [here](https://www.quantifiedcode.com/app/issue_class/62b0f5b7b69e4a2498568b32bfa30991) and also according to [this](https://www.youtube.com/watch?v=8bZh5LMaSmE&feature=youtu.be) video by Sandi Metz, anyday this would be preferred over the current one.